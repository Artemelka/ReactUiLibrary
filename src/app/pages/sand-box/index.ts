export { AnchorSandBox } from './anchor/anchor-sand-box';
export { ButtonSandBox } from './button/button-sand-box';
export { ButtonGroupSandBox } from './button/button-group-sand-box';
export { CardListSandBox } from './card-list/card-list-sand-box';
export { CheckboxSandBox } from './checkbox/checkbox-sand-box';
export { DropDownPanelSandBox } from './drop-down-panel/drop-down-panel-sand-box';
export { IconSandBox } from './icon/icon-sand-box';
export { InputSandBox } from './input/input-sand-box';
export { ProgressSandBox } from './progress/progress-sand-box';
export { SelectSandBox } from './select/select-sand-box';
export { TextSandBox } from './text/text-sand-box';

