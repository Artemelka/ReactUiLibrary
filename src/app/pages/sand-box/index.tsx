import React from 'react';
import { Route, Switch } from 'react-router';
import { ComponentsCollection } from '../constants';

export class SandBox extends React.Component {
    render() {
        return (
            <Switch>
                {ComponentsCollection.map(
                    ({ component, exact, url }, index) =>
                        <Route path={url} key={index} exact={exact} component={component} />
                )}
            </Switch>
        );
    }
}