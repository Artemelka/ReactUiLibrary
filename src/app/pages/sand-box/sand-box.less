.Sandbox {
    padding: 30px;
    text-align: left;

    &__item {
        margin: 50px 0;

        &--bg {
            &-white {
                background-color: #fff;
                padding: 5px;
            }
        }
    }
}
