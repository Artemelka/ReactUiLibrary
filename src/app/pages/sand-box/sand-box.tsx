import React from 'react';
import classNames from 'classnames/bind';

const style = require('./sand-box.less');
const cn = classNames.bind(style);

interface Props {
    bgWhite?: boolean;
    items: Array<React.ElementType | any>;
}

export const SandBox = (props: Props) => {
    const {items, bgWhite} = props;
    const itemClasses = cn('Sandbox__item', {
        'Sandbox__item--bg-white': bgWhite
    });

    return (
        <div className={cn('Sandbox')}>
            {items.map((item, index) => {
                const Element = item;
                return (
                    <div className={itemClasses} key={index}>
                        <Element />
                    </div>
                );
            })}
        </div>
    );
};
