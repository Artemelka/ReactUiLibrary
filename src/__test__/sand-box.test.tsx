import React from 'react';
import { shallow } from 'enzyme';
import { SandBox } from '../app/pages/sand-box/sand-box';

const items = [
    'test item'
];

describe('Test SandBox', () => {
    test('Rendering snapshot', () => {
        const wrapper = shallow(<SandBox items={items}/>);

        expect(wrapper).toMatchSnapshot();
    });
    test('Expect className modifier bg-white with prop bgWhite', () => {
        const wrapper = shallow(<SandBox bgWhite items={items}/>);

        expect(wrapper.find('.Sandbox__item')
            .hasClass('Sandbox__item--bg-white'))
            .toBe(true);
    });
});
