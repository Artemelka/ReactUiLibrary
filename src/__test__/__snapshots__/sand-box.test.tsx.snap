// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SandBox Rendering snapshot 1`] = `
<div
  className="Sandbox"
>
  <div
    className="Sandbox__item"
    key="0"
  >
    test item
  </div>
</div>
`;
